### ISSUE TITLE
Make `phf_codegen::Map` implement `std::fmt::Write`

### ISSUE THREAD

**Issue by althonos** [sfackler/rust-phf#133]
Hi @sfackler ! First of all, thanks a lot for this library, using it helped me get better performances for a structural biology software I am writing that needs to read *a lot* of parameter tables.

When I was writing my `build.rs` file however, I noticed that `phf_codegen::Map` does not implement the `std::fmt::Write` trait. Currently I have to write something like:
```rust
write!(&mut file, "static {}: {} =", name, ty)
    .and(map.build(&mut file))
    .and(write!(&mut file, ";\n"))
    .expect(&format!("could not write `{}` map", name));
```
But with `Write` implemented this would become:
```rust
write!(&mut file, "static {}: {} = {};\n", name, ty, map)
    .expect(&format!("could not write `{}` map", name));
```

 Is there a particular reason for this ? If the only reason is that you don't have time to implement that, I can open a PR.

**Comment by sfackler** [comment]
I think the thing to do here would be to make `build()` return a `Display` type rather than taking the writer in as an argument.

**Comment by althonos** [comment]
@sfackler : indeed, that looks like the most sensible option. Do you want me to work on a PR ?

**Comment by sfackler** [comment]
Sure!

**Comment by abonander** [comment]
Closed by #161
